rocky/rocky/views/upload_raw.py

```diff
@@ -15,11 +17,25 @@ class UploadRaw(OrganizationPermissionRequiredMixin, OrganizationView, FormView)
     template_name = "upload_raw.html"
     form_class = UploadRawForm
     permission_required = "tools.can_scan_organization"
+    mime_types = False
 
     def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
         if not self.organization:
             self.add_error_notification(RAW_ERRORS["no_org"])
+        if kwargs.get("mime_type"):
+            self.mime_types = unquote(kwargs["mime_type"])
+        elif kwargs.get("mime_types"):
+            self.mime_types = unquote(kwargs["mime_types"])
+
+    def get_initial(self):
+        """
+        Returns the initial data to use for forms on this view.
+        """
+        initial = super().get_initial()
+        if self.mime_types:
+            initial["mime_types"] = self.mime_types
+        return initial
```

Comment:
Can be done a bit simpler:

```suggestion
    def get_initial(self):
        """
        Returns the initial data to use for forms on this view.
        """
        initial = super().get_initial()
        if "mime_type" in self.kwargs:
            initial["mime_types"] = unquote(self.kwargs["mime_type"])
        return initial
```

I can't change unmodified lines in a suggestion, but the whole `setup` can go away, because `self.organization` is always set in `OrganisationView`.

Reply:
Ah, I did not know the kwargs where available on self here.